beetmoverscript/script.py

```diff
@@ -283,6 +301,70 @@ def list_bucket_objects(context, s3_resource, prefix):
                                                                                    locale, destinations, from_buildid))
 
 
+# move_partner_beets {{{1
+async def move_partner_beets(context, manifest):
+    artifacts_to_beetmove = context.artifacts_to_beetmove
+    beets = []
+
+    for locale in artifacts_to_beetmove:
+        for full_path_artifact in artifacts_to_beetmove[locale]:
+            source = artifacts_to_beetmove[locale][full_path_artifact]
+            destination = get_destination_for_private_repack_path(context, manifest,
+                                                                  full_path_artifact, locale)
+            beets.append(
+                asyncio.ensure_future(
+                    upload_to_s3(context=context, s3_key=destination, path=source)
+                )
+            )
+    await raise_future_exceptions(beets)
+
+
+def get_destination_for_private_repack_path(context, manifest, full_path, locale):
+    """Function to process the final destination path, relative to the root of
+    the S3 bucket. Depending on whether it's a private or public destination, it
+    performs several string manipulations.
+
+    Input: 'releng/partner/ghost/ghost-var/v1/linux-i686/ro/target.tar.bz2'
+    Possible ouput(s):
+        -> ghost/59.0b20-2/ghost-variant/v1/linux-i686/en-US/firefox-59.0b20.tar.bz2
+        -> pub/firefox/candidates/59.0b20-candidates/build2/partner-repacks/ghost/ghost-variant/v1/linux-i686/en-US/firefox-59.0b20.tar.bz2
+    """
+    # make sure we're calling this function from private-partner context
+    if not is_partner_action(context.action):
+        raise ScriptWorkerRetryException("Outside of private-partner context!")
+
+    # pretty name the `target` part to the actual filename
+    pretty_full_path = os.path.join(
+        os.path.dirname(full_path),
+        manifest['mapping'][locale][os.path.basename(full_path)]
+    )
+    # get rid of leading "releng/partner"
+    if pretty_full_path.startswith(PARTNER_LEADING_STRING):
+        pretty_full_path = pretty_full_path[len(PARTNER_LEADING_STRING):]
+
+    build_number = context.task['payload']['build_number']
+    version = context.task['payload']['version']
+
+    if is_partner_private_task(context):
+        elements = pretty_full_path.split('/')
+        identifier = '{version}-{build_number}'.format(version=version, build_number=build_number)
+        # we need need to manually insert the "version-buildno" identifier in
+        # between `partner` # and `partner-variant` to be consistent
+        elements.insert(1, identifier)
+        # TODO: potentially need to remove the `v1` from the path?
+        path = '/'.join(elements)
+        # XXX: temp hack until bug 1447673 is solved
+        if context.bucket == "dep":
```

Comment:
hm, won't the private and public uploads upload to the same place? Doesn't matter?

Reply:
Sadly it does. Absent this hack (which we need while we test on maple/staging env), the
* private tasks will try to upload to http://ftp.stage.mozaws.net/$partner/$partner-variant ... directly in the root of the bucket which I'm not sure a) we're allowed b) we should 
* public tasks will go under regular candidates/partner-repacks/$partner/$partner-variant